### ISSUE TITLE
Provide a way to initialize out-of-tree providers

### ISSUE THREAD

**Issue by zuzzas** [gardener/machine-controller-manager#513]
**What would you like to be added**:

I'd like to have a special gRPC method that is **called exactly once** on the start-up of MCM and subsequent initialization of out-of-tree providers.

Here's how the Cloud Provider Interface approaches this: https://github.com/kubernetes/cloud-provider/blob/master/cloud.go#L44-L47

**Why is this needed**:

For example, while using an in-house in-tree vSphere plugin, we've hit a limit on simultaneous login attempts. An Active Directory server was under a lot of duress before I've implemented persistent session storage.

Here's how I've hacked around this problem to enable persistent sessions:

* Implementation: https://github.com/flant/machine-controller-manager-provider-yandex/blob/50b1fdd35f55a09d1259f3a88b08630acdc7a02b/pkg/spi/spi.go
* Usage: https://github.com/flant/machine-controller-manager-provider-yandex/blob/50b1fdd35f55a09d1259f3a88b08630acdc7a02b/pkg/provider/core.go#L94

Basically, we add or get a session from a map. Ugly, but works.

**Comment by elankath** [comment]
There is no gRPC comm between MCM and MC controller. We are not very clear on the requirement. Considering that driver implementations are now handled in provider specific code, it is not clear what we should do now.

**Comment by aaronfern** [comment]
## Grooming discussion

There was no response to this issue, hence closing.
